### PR TITLE
ci: Enable `ccache` in Travis

### PR DESCRIPTION
**Pull request by urutva**
Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,15 +37,23 @@ matrix:
...
+        - sudo ln -s ../../bin/ccache arm-none-eabi-gcc
+        - sudo ln -s ../../bin/ccache arm-none-eabi-g++
+        - export PATH="$PATH:/usr/lib/ccache"
+        - popd
         # Install arm-none-eabi-gcc
         - pushd /home/travis/build && mkdir arm-gcc && cd arm-gcc
         - curl -L0 "https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2020q2/gcc-arm-none-eabi-9-2020-q2-update-x86_64-linux.tar.bz2?revision=05382cca-1721-44e1-ae19-1e7c3dc96118&la=en&hash=D7C9D18FCA2DD9F894FD9F3C3DC9228498FA281A" --output gcc-arm-none-eabi-9-2020-q2-update.tar.bz2
         - tar xf gcc-arm-none-eabi-9-2020-q2-update.tar.bz2
-        - export PATH="${PWD}/gcc-arm-none-eabi-9-2020-q2-update/bin:$PATH"
+        - export PATH="$PATH:${PWD}/gcc-arm-none-eabi-9-2020-q2-update/bin"
```
Comment (jainvikas8) [inline .travis.yml:56]: Why is this changed required?

Comment (urutva) [reply .travis.yml:56]: To let `ccache` get precedence over `arm-none-eabi-gcc/g++`. https://github.com/ARMmbed/mbed-os-example-blinky/pull/233/files/2eb5d98fe223fab9e6fb781828b51ea897100551#diff-354f30a63fb0907d4ad57269548329e3R48 ensures that `ccache` is called instead of `arm-none-eabi-gcc/g++`.

**Review by Patater — APPROVED** [review]
LGTM
